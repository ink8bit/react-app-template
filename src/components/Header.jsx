import React, { Component } from 'react';
import { Link } from '@reach/router';
import styled from '@emotion/styled';

import color from './../styles/color';
import offset from './../styles/offset';

const HeaderContainer = styled.header`
  background-color: ${color.HEADER_BG};
  padding: 10px ${offset.PADDING_COMMON};
  display: flex;
  justify-content: space-between;
  align-items: center;

  ul {
    list-style: none;
    padding: 0;
    background-color: ${color.HEADER_BG};
  }

  li {
    display: inline-block;
  }

  li + li {
    margin-left: 20px;
  }
`;

const HeaderLogo = styled.img`
  background-color: ${color.HEADER_LOGO_BG};
  width: 70px;
  height: 30px;
`;

const HeaderLink = styled(Link)`
  background-color: ${color.HEADER_BG};
  text-decoration: none;
  color: lightblue;

  &:hover {
    text-decoration: underline;
  }
`;

class Header extends Component {
  render() {
    return (
      <HeaderContainer>
        <HeaderLogo src="" alt="logo" title="logo" />
        <ul>
          <li>
            <HeaderLink to="/">Main</HeaderLink>
          </li>
          <li>
            <HeaderLink to="/search">Search</HeaderLink>
          </li>
        </ul>
      </HeaderContainer>
    );
  }
}

export default Header;
