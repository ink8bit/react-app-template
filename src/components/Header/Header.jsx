import React, { Component } from 'react';
import { Link } from '@reach/router';
import styled from '@emotion/styled';
import Loadable from 'react-loadable';

import color from '../../styles/color';
import offset from '../../styles/offset';
import Modal from './../Modal/Modal.jsx';

const HeaderContainer = styled.header`
  background-color: ${color.HEADER_BG};
  padding: 10px ${offset.PADDING_COMMON};
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: ${offset.HEADER_HEIGHT};

  li {
    display: inline-block;
  }

  li + li {
    margin-left: 20px;
  }
`;

const HeaderLogo = styled.img`
  background-color: ${color.HEADER_LOGO_BG};
  width: 70px;
  height: 30px;
`;

const HeaderLink = styled(Link)`
  text-decoration: none;
  color: ${color.LINK};

  &:hover {
    text-decoration: underline;
  }
`;

const loading = () => <span>loading...</span>;
const LoadableContent = Loadable({
  loader: () => import('./../Modal/ModalContent.jsx'),
  loading,
});

class Header extends Component {
  state = {
    showModal: false,
  };

  toggleModal = () => {
    this.setState({
      showModal: !this.state.showModal,
    });
  };

  render() {
    const { showModal } = this.state;

    const ModalLayout = showModal ? (
      <Modal>
        <LoadableContent toggleModal={this.toggleModal} name={name} />
      </Modal>
    ) : null;

    return (
      <HeaderContainer>
        <HeaderLogo onClick={this.toggleModal} src="" alt="logo" title="logo" />
        <ul>
          <li>
            <HeaderLink to="/">Main</HeaderLink>
          </li>
          <li>
            <HeaderLink to="/search">Search</HeaderLink>
          </li>
          {ModalLayout}
        </ul>
      </HeaderContainer>
    );
  }
}

export default Header;
