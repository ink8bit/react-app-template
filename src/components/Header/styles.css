.header {
  background-color: var(--secondary-background-color);
  padding: 10px var(--padding);
  display: flex;
  justify-content: center;
}

.item {
  display: inline-block;
}

.item + .item {
  margin-left: 20px;
}

.wrapper {
  display: flex;
  justify-content: space-between;
  align-items: center;
  flex-shrink: 0;
  width: 100%;
  max-width: 1000px;
}

.logo {
  background-color: #848484;
  width: 70px;
  height: 30px;
}

.link {
  text-decoration: none;
  color: var(--link-color);
}

.link:hover {
  text-decoration: underline;
}
