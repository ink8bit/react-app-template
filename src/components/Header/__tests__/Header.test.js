import React from 'react';
import { create } from 'react-test-renderer';

import Header from './../Header.jsx';

test('snapshot', () => {
  const c = create(<Header />);
  expect(c.toJSON()).toMatchSnapshot();
});
