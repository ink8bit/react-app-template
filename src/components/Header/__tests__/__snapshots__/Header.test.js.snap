// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<header
  className="header"
>
  <div
    className="headerWrapper"
  >
    <img
      alt="logo"
      className="headerLogo"
      onClick={[Function]}
      src=""
      title="logo"
    />
    <ul>
      <li
        className="headerItem"
      >
        <a
          aria-current="page"
          className="headerLink"
          href="/"
          onClick={[Function]}
        >
          Main
        </a>
      </li>
      <li
        className="headerItem"
      >
        <a
          className="headerLink"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </li>
    </ul>
  </div>
</header>
`;
